editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,356 @@
+    newVersionParameters.setDescription(descriptionTextField.getText());
+
+    // update the UI
+    createVersionButton.setText("Creating version....");
+    createVersionButton.setDisable(true);
+
+    // create a new version with the specified parameters
+    ListenableFuture<ServiceVersionInfo> newVersion = serviceGeodatabase.createVersionAsync(newVersionParameters);
+    newVersion.addDoneListener(() -> {
+      try {
+        // get the name of the created version and switch to it
+        ServiceVersionInfo createdVersionInfo = newVersion.get();
+        userCreatedVersion = createdVersionInfo.getName();
+        switchVersion(userCreatedVersion);
+
+        // hide the form from the UI as the sample only allows 1 version to be created
+        createVersionVBox.setVisible(false);
+        switchVersionButton.setDisable(false);
+
+      } catch (Exception ex) {
+        // if there is an error creating a new version, display an alert and reset the UI
+        if (ex.getCause().toString().contains("The version already exists")) {
+          new Alert(Alert.AlertType.ERROR, "A version with this name already exists.\nPlease enter a unique name").show();
+        } else {
+          new Alert(Alert.AlertType.ERROR, "Error creating new version").show();
+        }
+        createVersionButton.setText("Create version");
+        createVersionButton.setDisable(false);
+      }
+    });
+  }
+
+  /**
+   * Apply local edits to the service geodatabase and switch branch version.
+   */
+  @FXML
+  private void handleSwitchVersionButtonClicked() {
+
+    if (serviceGeodatabase.getVersionName().equals(defaultVersion)) {
+      // if the current version is the default version, switch to the user created version
+      switchVersion(userCreatedVersion);
+    } else if (serviceGeodatabase.getVersionName().equals(userCreatedVersion)) {
+      // if the current version is the user created version, check if there are local edits
+      if (!serviceGeodatabase.hasLocalEdits()) {
+        // if there are no local edits switch to the default version
+        switchVersion(defaultVersion);
+      } else {
+        // if local edits exist apply the edits
+        ListenableFuture<List<FeatureTableEditResult>> resultOfApplyEdits = serviceGeodatabase.applyEditsAsync();
+        resultOfApplyEdits.addDoneListener(() -> {
+          try {
+            // check if the server edit was successful
+            List<FeatureTableEditResult> edits = resultOfApplyEdits.get();
+            if (edits == null || edits.isEmpty()) {
+              new Alert(Alert.AlertType.ERROR, "Error applying edits on server").show();
+            } else {
+              // if the edits were successful, switch to the default version
+              switchVersion(defaultVersion);
+            }
+          } catch (InterruptedException | ExecutionException e) {
+            new Alert(Alert.AlertType.ERROR, "Error applying edits on server").show();
+          }
+        });
+      }
+    }
+  }
```

Comment:
I think a bit of whitespace between the blocks would help make this section a little easier to read:

```suggestion
    if (serviceGeodatabase.getVersionName().equals(defaultVersion)) {
      // if the current version is the default version, switch to the user created version
      switchVersion(userCreatedVersion);
      
    } else if (serviceGeodatabase.getVersionName().equals(userCreatedVersion)) {
      // if the current version is the user created version, check if there are local edits
      if (!serviceGeodatabase.hasLocalEdits()) {
        // if there are no local edits switch to the default version
        switchVersion(defaultVersion);
      
      } else {
        // if local edits exist apply the edits
        ListenableFuture<List<FeatureTableEditResult>> resultOfApplyEdits = serviceGeodatabase.applyEditsAsync();
        resultOfApplyEdits.addDoneListener(() -> {
          try {
           
            // check if the server edit was successful
            List<FeatureTableEditResult> edits = resultOfApplyEdits.get();
            if (edits == null || edits.isEmpty()) {
              new Alert(Alert.AlertType.ERROR, "Error applying edits on server").show();
            
            } else {
              // if the edits were successful, switch to the default version
              switchVersion(defaultVersion);
            }
          } catch (InterruptedException | ExecutionException e) {
            new Alert(Alert.AlertType.ERROR, "Error applying edits on server").show();
          }
        });
      }
    }
  }
```

Reply:
Agreed, looks more readable with those spaces